opal/corelib/io.rb

```diff
@@ -3,6 +3,25 @@ class IO
   SEEK_CUR = 1
   SEEK_END = 2
 
+  attr_reader :eof
+  attr_reader :lineno
+
+  def initialize
+    @eof = false
+    @lineno = 0
+  end
+
+  def read
+    if @eof
+      ''
+    else
+      res = File.read(@path)
```

Comment:
Yes, that was the idea, but on second though I think `File#read` should be implemented directly in the node specific stdlib additions, i.e. in [this file](https://github.com/opal/opal/blob/master/stdlib/nodejs/file.rb) where `File.read` is already implemented. That's because I realized that reading files isn't very popular in browsers (which is Opal's main target) and the implementation would be node-specific anyway. 

Also should be pretty straightforward using `fs.readFileSync` which supports all kinds of offset/length parameters: https://nodejs.org/dist/latest-v4.x/docs/api/fs.html#fs_fs_readsync_fd_buffer_offset_length_position